### PR TITLE
fetch attributes based on aggregates

### PR DESCRIPTION
**Pull request by gibkigonzo**
related: https://github.com/DivanteLtd/vue-storefront/pull/4001 https://github.com/DivanteLtd/mage2vuestorefront/pull/99

Description:
In general this change allow to fetch attributes with needed options
for example: `{'color': [49]}` => will return color attribute with option that has id 49
It's used only for products (and response is transformed into attribute_metadata), but we can expose it as endpoint if needed.

### CODE REVIEW

File: src/api/attribute/service.ts [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+        }
+      })
+
+    // if all requested attributes are in cache then we can return here
+    if (!attributeCodes.length) {
+      return resolve(cachedAttributeList)
+    }
+
+    // fetch attributes for rest attributeCodes
+    request({
```
Comment (pkarw) [inline src/api/attribute/service.ts]: we could have used the `elasticsearch-js` client, but it should work that way as well - I guess

File: src/api/catalog.ts [hunk 2]
```diff
@@ -125,11 +156,14 @@ export default ({config, db}) => async function (req, res, body) {
         let resultProcessor = factory.getAdapter(entityType, indexName, req, res)
 
         if (!resultProcessor) { resultProcessor = factory.getAdapter('default', indexName, req, res) } // get the default processor
-
         if (entityType === 'product') {
-          resultProcessor.process(_resBody.hits.hits, groupId).then((result) => {
+          resultProcessor.process(_resBody.hits.hits, groupId).then(async (result) => {
             _resBody.hits.hits = result
             _cacheStorageHandler(config, _resBody, reqHash, tagsArray)
+            if (_resBody.aggregations) {
```
Comment (pkarw) [inline src/api/catalog.ts]: This way this feature won't work for graphql endpoint. graphql resolvers are using processors. we should finda way way to have it supported by graphql resolvers as well

File: src/api/attribute/service.ts (hunk 1 above)
Comment (gibkigonzo) [reply src/api/attribute/service.ts]: done :)

File: src/api/catalog.ts (hunk 2 above)
Comment (gibkigonzo) [reply src/api/catalog.ts]: I've changed it a little bit and add graphql support https://github.com/DivanteLtd/vue-storefront-api/pull/399/files#diff-fe1dd7c73a325a992278d55bfa329092R72

**Pull request merged**